numba/dictobject.py

```diff
@@ -657,6 +657,19 @@ def imp(key, value):
     return imp
 
 
+@overload(bool)
+def impl_bool(d):
+    """bool(dict)
+    """
+    if not isinstance(d, types.DictType):
```

Comment:
In comparison to the other implementations this has slightly different logic. How about the following instead:

```
if isinstance(d, types.DictType):
        def impl(d):
            return _dict_length(d) > 0

        return impl
```

Reply:
Done.